com.microsoft.mrtk.accessibility/Subsystems/MRTK/MRTKAccessibilitySubsystem.cs

```diff
@@ -40,10 +41,153 @@ class AccessibilityProvider : Provider
 
             public AccessibilityProvider() : base()
             {
+                describableObjects.Add(ObjectClassification.People, new List<GameObject>());
+                describableObjects.Add(ObjectClassification.Places, new List<GameObject>());
+                describableObjects.Add(ObjectClassification.Things, new List<GameObject>());
+                describableObjects.Add(ObjectClassification.UserInterface, new List<GameObject>());
+                describableObjects.Add(ObjectClassification.Background, new List<GameObject>());
+
                 Config = XRSubsystemHelpers.GetConfiguration<AccessibilitySubsystemConfig, MRTKAccessibilitySubsystem>();
                 invertTextColor = Config.InvertTextColor;
             }
 
+            #region Describable object management
+
+            private readonly Dictionary<ObjectClassification, List<GameObject>> describableObjects =
+                new Dictionary<ObjectClassification, List<GameObject>>();
+
+            /// <inheritdoc/>
+            internal override bool TryGetDescribableObjects(ObjectClassification classifications, ReaderView readerView, float maxDistance, List<GameObject> objectList)
+            {
+                if (maxDistance <= 0)
+                {
+                    Debug.LogError("The distance from the camera to the objects cannot be less than or equal to zero.");
+                    return false;
+                }
+
+                objectList.Clear();
+
+                AssembleDescribableObjects(classifications, objectList);
+                FilterDescribableObjects(readerView, maxDistance, objectList);
+
+                return true;
+            }
+
+            /// <inheritdoc/>
+            public override bool TryRegisterDescribableObject(GameObject gameObj, ObjectClassification classification)
+            {
+                // Make sure the specified game object has not been previously registered in any classification.
```

Comment:
Do you see a time where the possible classifications might have some overlap and one object _could_ be multiple? Or is that not something that would happen?

Reply:
at this point, i cannot think of a valid reason to have any overlap. a thing is not a place, a person is not ui, etc. :)

I _can_ see where the background category may be confusing and i am more than willing to remove it until it proves useful.

Reply:
removed background.